### ISSUE TITLE
setShowWeekends(0) does not change column colors

### ISSUE THREAD

**Issue by silviagreen** [jsGanttImproved/jsgantt-improved#255]
Hello,
I notices that when the setShowWeekends function is set to false, the columns which should highlight the weekend remain:
![immagine](https://user-images.githubusercontent.com/2613299/64790484-75c1c080-d576-11e9-8ef0-4ee1f9c8b83b.png)

Steps to reproduce the behavior (tried on Chrome and Firefox at the moment):
1. Create a Gantt
2. execute g.setShowWeekends(0);

Expected behaviour: the greyed column should disappear.

I also notices that in jsgantt.js (v2.0.10), the color of the cell is 'decided' by this control:

`for (j = 0; j < vNumCols - 1; j++) {
  if (this.vFormat == 'day' && ((j % 7 == 4) || (j % 7 == 5)))
               vCellFormat = 'gtaskcellwkend';
         else
                vCellFormat = 'gtaskcell';
        draw_utils_1.newNode(vTmpRow_1, 'td', null, vCellFormat, '\u00A0\u00A0');
}`

(lines: 540, 575, 629).

Changing this control as follows seems to remove the problem:

` for (j = 0; j < vNumCols - 1; j++) {
                                            if (this.vShowWeekends && this.vFormat == 'day' && ((j % 7 == 4) || (j % 7 == 5)))
                                                vCellFormat = 'gtaskcellwkend';
                                            else
                                                vCellFormat = 'gtaskcell';
                                            draw_utils_1.newNode(vTmpRow_1, 'td', null, vCellFormat, '\u00A0\u00A0');
                                        }`

PS: sorry for the code styling, I cannot understand how to add new lines :(

**Comment by mariohmol** [comment]
something seems wrong here..

check the demo https://jsganttimproved.github.io/jsgantt-improved/demo.html

there is a option `Show Weekenddays` and if check / uncheck there you will see the weekends going away

**Comment by mariohmol** [comment]
included in latest version! thanks

**Comment by reregaga** [comment]
@mariohmol, I tried the latest version (2.8.10) from npm and it keeps colorizing cells when  `vShowWeekends` are disabled.

![Image](https://github.com/user-attachments/assets/66ca32e7-57d8-451c-83be-6b0953b59281)

```html
<!DOCTYPE html><html>
<head>
  <meta charset="utf-8"><meta name="viewport" content="width=device-width"><title>JS Bin</title>
  <link href="https://cdn.jsdelivr.net/npm/jsgantt-improved@2.8.10/dist/jsgantt.css" rel="stylesheet"></link>
  <script src="https://cdn.jsdelivr.net/npm/jsgantt-improved@2.8.10/dist/jsgantt.js"></script>
</head>
<body>
<div style="position:relative" class="gantt" id="GanttChartDIV"></div>
<script>
let g = new JSGantt.GanttChart(document.getElementById('GanttChartDIV'), 'day');

g.setOptions({
  vShowWeekends: 0,
  
  vDateTaskDisplayFormat: 'day dd month yyyy',
  vDayMajorDateDisplayFormat: 'mon yyyy - Week ww',
  vWeekMinorDateDisplayFormat: 'dd mon',
  vFormatArr: ['Hour', 'Day', 'Week', 'Month', 'Quarter'],
  vLang: 'en', vQuarterColWidth: 36, vScrollTo: 'today', vTooltipDelay: 1,
  vShowCost: 0, vShowComp: 0, vShowDur: 0, vShowStartDate: 0, vShowEndDate: 0,
  vShowPlanStartDate: 0, vShowPlanStartDate: 0, vShowRes: 0, vShowEndWeekDate: 0,
});

g.AddTaskItemObject({
  pID: 1, pName: "Name",
  pStart: new Date(),
  pEnd: new Date(Date.now() + 15*24*60*60*1000),
  pClass: "ggroupblack",
});
g.Draw();
</script>
</body></html>
```
Live: https://jsbin.com/pecolofonu/1/edit?html,output

Also, when `vShowWeekends` enabled, the cells are not colorized when the `Hour` format is selected... So I don't know if it makes sense to create a separate feature-request and I'm posting it here.

Anyway, thank you. I'm choosing a lib for a home project and yours has impressed me with the number of features, it is a notable lib.

**Comment by mariohmol** [comment]
Fixed in PR #395 (merged). The root cause was that `drawColsChart` in `draw.ts` used strict `!== false` to check `vShowWeekends`, so passing the numeric `0` (as documented) evaluated `0 !== false → true` and kept weekend cells colored. Changed both checks to a truthy test (`this.vShowWeekends`), consistent with the rest of the file. Covered by unit tests and a Playwright e2e test.
